sentinel.go

```diff
@@ -205,6 +205,31 @@ func (sc *Sentinel) Do(a Action) error {
 	return sc.clients[sc.primAddr].Do(a)
 }
 
+// DoSecondary is like Do but executes the Action on a random replica if possible.
+//
+// For DoSecondary to work, replicas must be configured with replica-read-only
+// enabled, otherwise calls to DoSecondary may by rejected by the replica.
+//
+// NOTE it's possible that in between Do being called and the Action being
+// actually carried out that there could be a failover event. In that case, the
+// Action will likely fail and return an error.
+func (sc *Sentinel) DoSecondary(a Action) error {
+	sc.l.RLock()
+	var addr string
+	for addr = range sc.clients {
+		if addr != sc.primAddr {
+			break
+		}
+	}
+	sc.l.RUnlock()
+
+	c, err := sc.Client(addr)
```

Comment:
Right now this `DoSecondary` method is just a convenience helper, it doesn't do anything that the user couldn't do themselves. _But_ if you modified how `Client` works a bit you could remove an `RLock()`/`RUnlock()` in the path. Basically, move most of `Client`'s code into a new method, `clientInner` or something like that, and have that new method also accept an empty address string which it interprets to mean "random secondary". Then both `Client` and `DoSecondary` can call that directly. It might be worthwhile to have the new `Client` method check for being given an empty address? Not 100% on that.

Reply:
Done